langchain4j-core/src/main/java/dev/langchain4j/rag/content/Content.java

```diff
@@ -20,47 +18,35 @@
  * @see ContentAggregator
  * @see ContentInjector
  */
-public class Content {
-
-    private final TextSegment textSegment;
-
-    public Content(String text) {
-        this(TextSegment.from(text));
-    }
+public interface Content {
 
-    public Content(TextSegment textSegment) {
-        this.textSegment = ensureNotNull(textSegment, "textSegment");
-    }
+    TextSegment textSegment();
 
-    public TextSegment textSegment() {
-        return textSegment;
+    static Content from(String text) {
+        return new TextContent(text);
     }
 
-    @Override
-    public boolean equals(Object o) {
-        if (this == o) return true;
-        if (o == null || getClass() != o.getClass()) return false;
-        Content that = (Content) o;
-        return Objects.equals(this.textSegment, that.textSegment);
+    static Content from(TextSegment textSegment) {
+        return new TextContent(textSegment);
     }
 
-    @Override
-    public int hashCode() {
-        return Objects.hash(textSegment);
-    }
+    record TextContent(
```

Comment:
Not sure about the name... few reasons:
1. we already have `TextContent` in another package (not a major issue)
2. while it contains only text content right now, it might contain other content types in the future. Consider tables or images in PDF file with text captions. This pieces of content are indivisible and will require both text and image pair to be stored/retrieved together.

What about `DefaultContent`?

Reply:
Renamed